src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -94,73 +87,124 @@ public void setup() throws InterruptedException, ServiceBusException, JSONExcept
+
+        verifyExistingCaseIsUpdatedWithExceptionRecordData(caseDetails, exceptionRecord);
+    }
+
+    private void invokeCallbackEndpointForLinkingDocsToCase(CaseDetails caseDetails, CaseDetails exceptionRecord) {
+        Map<String, Object> caseData = exceptionRecord.getData();
+        caseData.put("attachToCaseReference", String.valueOf(caseDetails.getId()));
+
         CallbackRequest callbackRequest = CallbackRequest
             .builder()
-            .eventId("attach_case")
-            .caseDetails(exceptionRecordCase)
-            .caseDetailsBefore(existingCase)
+            .eventId("attachToExistingCase")
+            .caseDetails(exceptionRecord)
             .build();
 
-        Response callbackResponse = RestAssured
+        RestAssured
             .given()
+            .relaxedHTTPSValidation()
             .baseUri(testUrl)
             .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
             .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Orchestrator Functional test")
             .body(callbackRequest)
             .when()
-            .post("/callback/{type}", "attach_case")
-            .andReturn();
-
-        //then
-        assertThat(callbackResponse.getStatusCode()).isEqualTo(200);
-        verifyExistingCaseIsUpdatedWithExceptionRecordCaseDetails();
+            .post("/callback/attach_case");
     }
 
-    private boolean isExceptionRecordCreated(UUID randomPoBox) {
+    private Optional<CaseDetails> lookUpExceptionRecord(UUID randomPoBox) {
         List<CaseDetails> caseDetailsList = caseSearcher.search(
             SampleData.JURSIDICTION,
             EnvelopeEventProcessor.EXCEPTION_RECORD_CASE_TYPE,
             ImmutableMap.of(
                 "case.poBox", randomPoBox.toString()
             )
         );
-
-        exceptionRecordCase = caseDetailsList.isEmpty() ? null : caseDetailsList.get(0);
-        return exceptionRecordCase != null;
+        return caseDetailsList.stream().findFirst();
     }
 
-    private void verifyExistingCaseIsUpdatedWithExceptionRecordCaseDetails() {
+    private Boolean isExceptionRecordAttachedToTheCase(CaseDetails caseDetails, int expectedScannedDocsSize) {
+
         CaseDetails updatedCase = caseRetriever.retrieve(
-            existingCase.getJurisdiction(),
-            String.valueOf(existingCase.getId())
+            caseDetails.getJurisdiction(),
+            String.valueOf(caseDetails.getId())
         );
 
         List<ScannedDocument> updatedScannedDocuments = getScannedDocumentsForSupplementaryEvidence(updatedCase);
 
-        List<ScannedDocument> scannedDocuments = getScannedDocumentsForExceptionRecord(exceptionRecordCase);
+        return updatedScannedDocuments.size() == expectedScannedDocsSize;
+    }
 
-        assertThat(scannedDocuments).isNotEmpty();
-        assertThat(scannedDocuments.size()).isEqualTo(1);
+    private void verifyExistingCaseIsUpdatedWithExceptionRecordData(
+        CaseDetails caseDetails,
```

Comment:
Its too tempting in this method to use the old caseDetails to check for the documents.
It would be better to pass in 
```
           caseDetails.getJurisdiction(),
            String.valueOf(caseDetails.getId())
```
and force the method/maintainer to look the case up

Also Since the method is called `ExceptionRecordData` you should just pass that in too 
Id have expected a method signature similar to 
verifyCaseHasDocuments( jurisdiction, caseRef, List<ScannedDocuments> )
then build the list in the test to check that the updated test contains the documents you are expecting.